wsinfer/qupath.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import sys
+import json
+from pathlib import Path
+
+
+def add_image_and_geojson(
+    qupath_proj: QuPathProject, *, image_path: Path | str, geojson_path: Path | str
+) -> None:
+    with open(geojson_path) as f:
+        # FIXME: check that a 'features' key is present and raise a useful error if not
+        geojson_features = json.load(f)["features"]
+
+    entry = qupath_proj.add_image(image_path)
+    # FIXME: test that the 'load_geojson' function exists. If not, raise a useful error
+    try:
+        entry.hierarchy.load_geojson(geojson_features)  # type: ignore
+    except Exception as e:
+        print(f"Failed to run load_geojson function with error:: {e}")
+
+
+# Store a list of matched slides and geojson files. Linking the slides and geojson in
+# this way prevents a potential mismatch by simply listing directories and relying on
+# the order to be the same.
+
+
+def make_qupath_project(wsi_dir, results_dir):
+    try:
+        from paquo.projects import QuPathProject
+    except Exception as e:
+        print(
+            f"""Cannot find QuPath with error:: {e}.
+QuPath is required to use this functionality but it cannot be found.
+If QuPath is installed, please use define the environment variable
+PAQUO_QUPATH_DIR with the location of the QuPath installation.
+If QuPath is not installed, please install it from https://qupath.github.io/."""
+        )
+        sys.exit(1)
+
+    print("Found QuPath successfully!")
+    QUPATH_PROJECT_DIRECTORY = "QuPathProject"
```

Comment:
can we move the qupath output directory into the results_dir? maybe

```python
results_dir / "model-outputs-qupath"
```

Reply:
Yes! That would be a better way to store pretty much all the results in one directory